WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/ShippingLabelPackageDetailsViewModel.swift

```diff
@@ -17,11 +18,11 @@ final class ShippingLabelPackageDetailsViewModel: ObservableObject {
 
     /// Products contained inside the Order and fetched from Core Data
     ///
-    private var products: [Product] = []
+    @Published private var products: [Product] = []
 
     /// ProductVariations contained inside the Order and fetched from Core Data
     ///
-    private var productVariations: [ProductVariation] = []
+    @Published private var productVariations: [ProductVariation] = []
```

Comment:
A thing that we should keep in mind when working with Combine, is that not always declaring a variable as `@Published` is a benefit since the variables will be observed also by SwiftUI (because the view model is declared as `ObservableObject`), so every change will affect the UI performances. In this specific case, there shouldn't be any performance issues, but that's something to keep in mind for the future.

Reply:
I agree! We should totally make a property a publisher only when needed. In this case, I need these to be publishers in order to observe changes in their values.